### PR TITLE
{CI} Add policy check to Security Tools

### PR DESCRIPTION
**Pull request by wangzelin007**
[Policy Check](https://policheck.microsoft.com/)
PoliCheck contains sensitive terms in over 100 languages and provides guidance to determine the appropriate action depending on context. 
The term data also provides explanations of why certain terms are sensitive or unacceptable.
image

### CODE REVIEW

**Review by kairu-ms — APPROVED** [review]

File: azure-pipelines.yml [hunk 1]
```diff
@@ -57,12 +57,20 @@ jobs:
       toolMajorVersion: V2
       suppressionsFile: './scripts/ci/credscan/CredScanSuppressions.json'
       toolVersionV2: '2.1.17'
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-policheck.PoliCheck@2
```
Comment (jiasli) [inline azure-pipelines.yml:61]: Can this task be run in parallel with CredScan, given CredScan is already the slowest task?

**Comment by yonzhan** [comment]
Add policy check

File: azure-pipelines.yml (hunk 1 above)
Comment (jiasli) [inline azure-pipelines.yml:61]: Is there any document for this ADO task?

File: azure-pipelines.yml [hunk 2]
```diff
@@ -57,12 +57,20 @@ jobs:
...
+    displayName: 'Run PoliCheck'
+    inputs:
+      targetType: F
```
Comment (jiasli) [inline azure-pipelines.yml:64]: What does `F` mean here?

Comment (wangzelin007) [reply azure-pipelines.yml:64]: F: Path|Filename - specify root folder | file to be scanned. 
FL:FILE - specify a list of file containing folder | file paths to be scanned.

File: azure-pipelines.yml (hunk 1 above)
Comment (wangzelin007) [reply azure-pipelines.yml:61]: https://eng.ms/docs/security-compliance-identity-and-management-scim/security/azure-security/cloudai-security-fundamentals-engineering/security-integration/guardian-wiki/sdl-azdo-extension/policheck-build-task

Comment (wangzelin007) [reply azure-pipelines.yml:77]: Sure, It's just that we need to add one more PostAnalysis task to check the result of the PoliCheck

File: azure-pipelines.yml [hunk 3]
```diff
@@ -57,16 +57,36 @@ jobs:
...
+  pool:
+    name: ${{ variables.windows_pool }}
+  steps:
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-policheck.PoliCheck@2
+    displayName: 'Run Policy Check'
+    inputs:
+      targetType: F
+      result: PoliCheckResult.xml
+
+  - task: securedevelopmentteam.vss-secure-development-tools.build-task-postanalysis.PostAnalysis@2
```
Comment (wangzelin007) [inline azure-pipelines.yml:83]: If using PostAnalysis@1, an error will be reported: `Could not find any reports, please choose a tool.`
But PostAnalysis@2 works fine.

File: azure-pipelines.yml (hunk 2 above)
Comment (jiasli) [reply azure-pipelines.yml]: The description is still pretty unclear.

File: azure-pipelines.yml (hunk 3 above)
Comment (jiasli) [inline azure-pipelines.yml:83]: We may have some documents for how this YAML job is generated.

**Review by jiasli — APPROVED** [review]

File: azure-pipelines.yml [hunk 4]
```diff
@@ -57,16 +57,36 @@ jobs:
...
+    displayName: 'Post Analysis'
+    inputs:
+      GdnBreakAllTools: false
```
Comment (wangzelin007) [inline azure-pipelines.yml:86]: Gdn means [guardian](https://eng.ms/docs/security-compliance-identity-and-management-scim/security/azure-security/cloudai-security-fundamentals-engineering/security-integration/guardian-wiki/microsoft-guardian/guardian-commands/break).

File: azure-pipelines.yml (hunk 3 above)
Comment (wangzelin007) [reply azure-pipelines.yml:83]: I will use the classic editor to create a pipeline without YAML.
In this way you can check the meaning of each parameter directly on the UI.
After testing, you can use the **View YAML** function to automatically convert to yaml format.
image
image

File: azure-pipelines.yml (hunk 4 above)
Comment (wangzelin007) [reply azure-pipelines.yml:86]: All documents can be easily found by clicking the button on the UI
image

**Pull request merged**